src/main/java/com/hotels/service/tracing/zipkintohaystack/ZipkinController.java

```diff
@@ -79,6 +79,7 @@ public ZipkinController(SpanForwarder... spanForwarders) {
         return serverRequest
                 .bodyToMono(byte[].class)
                 .flatMapIterable(decodeList(decoder))
+                .filter(span -> span.traceId() != null) // TODO: add logging here? or metric with discarded/invalid spans
```

Comment:
What happened to trigger this?

Reply:
This came up because of the use of the trace id for the partition key in both kafka and kinesis.
But I think this is being overcautious. I've never came across a trace with an empty trace id, and that field is mandatory as per https://zipkin.io/zipkin-api/#/default/post_spans.

I'll remove this check.

Reply:
No, I think it's a good idea, just wondered if you'd seen it happen :)